### DISCUSSION TITLE
Enum: can I allow undefined value in an enum ?

### DISCUSSION THREAD

**Discussion by maxime4000** [prisma/prisma#18051] [Q&A] [answered]
I'm registering user's device so I can store it's access token. Let's said I have this enum {WEB, IOS, ANDROID}, if for whatever reason I received something else, let's said I'm making call from PostMan and I have not specify POSTMAN in the enum, can I still allow POSTMAN to be add to the field even if the enum doesn't include POSTMAN ? 

Am I better with a string at this point or enum would still allow this ?

**Comment by ludralph** [comment, marked as answer]
Hi @maxime4000 👋 

If you are using an Enum to represent the possible values for the device type, then by definition, only those specific values can be accepted as valid input. So, you will get an error if you try to store a device type value that is not included in the enum.

In your scenario, if you want to allow for a value like "POSTMAN" to be added to the field, then using an enum is not a good choice. Instead, you could use a string to represent the device type, allowing any value to be stored. This would allow you to accept any device and store its access token.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
